src/ccutil/fileio.cpp

```diff
@@ -113,16 +122,52 @@ bool File::DeleteMatchingFiles(const char* pattern) {
  return all_deleted;
 }
 #else
-bool File::DeleteMatchingFiles(const char* pattern) {
+bool File::DeleteMatchingFiles(const char* directory, const char* pattern) {
+  bool all_deleted = true;
+
+#ifdef __ANDROID__
+  DIR *dir;
+  struct dirent *dp;
+  regex_t match;
+  int res;
+  char msg[100];
+  const char *full_path;
+
+  regcomp(&match, pattern, REG_BASIC);
+  dir = opendir(directory);
+
+  if (dir != NULL) {
+    while ((dp = readdir(dir)) != NULL) {
+      if (!(res = regexec(&match, dp->d_name, 0, NULL, 0))) {
+        full_path = File::JoinPath(directory, dp->d_name).c_str();
+        all_deleted &= File::Delete(full_path);
+      }
+      else if (res == REG_NOMATCH) {
+        // No match, noop or warn?
+      }
+      else {
+        // Print error message?
+
+        //regerror(res, &match, msg, sizeof(msg));
+        //fprintf(stderr, "Regex match failed: %s\n", msg);
+        //exit(1);
```

Comment:
In case these are kept, what's the way to go. I'm assuming exiting here isn't the desired outcome.